pkg/api/auth/auth_handler.go

```diff
@@ -0,0 +1,105 @@
+package auth
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"net/http"
+
+	"github.com/rancher/harvester/pkg/auth/jwe"
+	"github.com/rancher/harvester/pkg/auth/jwt"
+	"github.com/rancher/harvester/pkg/config"
+
+	dashboardauthapi "github.com/kubernetes/dashboard/src/app/backend/auth/api"
+	"github.com/pkg/errors"
+	"k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/client-go/tools/clientcmd/api"
+)
+
+const (
+	jwtServiceAccountClaimSubject = "sub" //https://github.com/kubernetes/kubernetes/blob/master/pkg/serviceaccount/claims.go#L64
```

Comment:
prefer to use https://github.com/kubernetes/kubernetes/blob/51ffb495f752fadba8781cc71d4708abd301de75/pkg/serviceaccount/claims.go#L64